src/poetry/puzzle/solver.py

```diff
@@ -99,7 +99,7 @@ def solve_in_compatibility_mode(
         depths = []
         for override in overrides:
             self._provider.debug(
-                "<comment>Retrying dependency resolution "
+                "<comment>Retrying dependency resolution "  # noqa: PIE803
```

Comment:
If we're going to embrace the style changes proposed by flake8-pie, I think we should do so consistently here rather than using `# noqa`.

Reply:
I realize now I'm not sure where the `debug()` method on `_provider` comes from -- if there's no built in interpolation, fair enough (but add a comment explaining why). Seems like this might be a fairly coarse lint.